src/bindings/c/src/CMakeLists.txt

```diff
@@ -17,8 +17,12 @@ target_link_libraries(${TARGET_NAME} PRIVATE openvino)
 target_include_directories(${TARGET_NAME} PUBLIC
     $<BUILD_INTERFACE:${InferenceEngine_C_API_SOURCE_DIR}/include>)
 
-if(NOT BUILD_SHARED_LIBS)
-    target_compile_definitions(${TARGET_NAME} PUBLIC OPENVINO_STATIC_LIBRARY)
+find_package(OpenCL QUIET)
+if(NOT OpenCL_FOUND)
+    MESSAGE(WARNING "OpenCL is disabled or not found, ${TARGET_NAME} will be built without OpenCL support. Install OpenCL.")
+    target_link_libraries(${TARGET_NAME} PRIVATE openvino)
+else()
+    target_link_libraries(${TARGET_NAME} PRIVATE openvino ${OpenCL_LIBRARY})
```

Comment:
We cannot depend on OpenCL in common library..

Reply:
In order not to have dependency on libOpenCL.so, we have to create common method which are exported from the library. E.g. create a remote content based on the specific map of parameters, while helper functions (like in C++ API) must be inline (otherwise we have dependency on the backend library).

And in this case we cannot use C++ helpers methods like you used in .c file

Reply:
Eh... So, using inline function will not check the libOpenCL.so dependency? In .c file, what should I use?